### ISSUE TITLE
S3 list objects whitespace in prefix generates a bad signature

### ISSUE THREAD

**Issue by Derya** [akka/alpakka#760]
Hey all, it looks like if I call `client.listBucket(myBucket, Option(somePrefixWithASpace))`, the library generates the signature incorrectly. 
```
akka.stream.alpakka.s3.S3Exception: The request signature we calculated does not match the signature you provided. Check your key and signing method.
```

**Comment by ennru** [comment]
Even reported in #719

**Comment by ennru** [comment]
Fixed by #771
